### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by Mikearaya**
Hi
on this pull request, I have tested all the cases for the enumerable methods I've created
1. #my_each 
2. #my_each_with_index
3. #my_select (#my_find_all, #my_filter)
4. #my_all? 
5. #my_any?
6. #my_none?
7. #my_count 
8. #my_map

### I'm kindly requesting your review so I can proceed and merge all tests to develop and `master`

**Thank you** :bowing_man:

### CODE REVIEW

File: spec/iterator_spec.rb [hunk 1]
```diff
@@ -0,0 +1,220 @@
+require './iterators.rb'
+describe 'Iterators' do
+  let(:number) { [1, 2, 3, 4, 2] }
+  let(:string) { %w[cat bear rat] }
+  context '#my_all?' do
+    it 'Return true when no block is given' do
+      expect(number.my_all?).to eq true
+    end
```
Comment (mosesogwo) [inline spec/iterator_spec.rb]: Include test for when no block is given and there is at least one element that is false or nil. In that case, it should return false

File: spec/iterator_spec.rb [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+
+    it 'Return false when some elements do not match pattern' do
+      expect(string.my_all?(/t/i)).not_to eq true
+    end
+  end
+
+  context '#my_any?' do
+    it 'Return true when no block is given' do
+      expect(number.my_any?).to eq true
+    end
```
Comment (mosesogwo) [inline spec/iterator_spec.rb]: When no block is given and none of the elements is a truthy, it should return false. Add test for that scenario

File: spec/iterator_spec.rb [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+
+    it 'number of items return true when block execute' do
+      expect(number.my_count(&:even?)).to eq 3
+    end
+  end
+
+  context '#my_filter should Return ' do
+    it 'items that match block expression' do
+      expect(number.my_filter(&:even?)).to all(be_even)
```
Comment (mosesogwo) [inline spec/iterator_spec.rb]: A better check will be to expect the exact array of even numbers. Any other set of even numbers will pass this test... but we don't just want it to return any set of even numbers.

File: spec/iterator_spec.rb [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+    end
+
+    it 'enumerator when no block is given' do
+      expect(number.my_filter).to be_instance_of(Enumerator)
+    end
+  end
+
+  context '#my_select should Return ' do
```
Comment (mosesogwo) [inline spec/iterator_spec.rb]: This is an alias for my_filter. One method test for all the aliases should be sufficient since the method definition is the same.

File: spec/iterator_spec.rb [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+    it 'items that match block expression' do
+      expect(number.my_select(&:even?)).to all(be_even)
+    end
+
+    it 'enumerator when no block is given' do
+      expect(number.my_select).to be_instance_of(Enumerator)
+    end
+  end
+
+  context '#my_find_all should Return ' do
```
Comment (mosesogwo) [inline spec/iterator_spec.rb]: This is an alias for my_filter. One method test for all the aliases should be sufficient since the method definition is the same.

File: spec/iterator_spec.rb [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+    it 'items that match block expression' do
+      expect(number.my_find_all(&:even?)).to all(be_even)
+    end
+
+    it 'enumerator when no block is given' do
+      expect(number.my_find_all).to be_instance_of(Enumerator)
+    end
+  end
+
+  context '#my_inject should Return ' do
```
Comment (mosesogwo) [inline spec/iterator_spec.rb:94]: inject and reduce are aliases, one test should be sufficient.

**Review by mosesogwo — CHANGES_REQUESTED** [review]
**Hi @Mikearaya, good job with your tests.:+1:. Please see my comments on your spec files**


_**Kindly make those changes and submit another code review request.**_

**Any questions? You can reach me on Slack (Moses Ogwo)**

**Cheers:stars:**

### Status: Changes Requested.

File: spec/iterator_spec.rb (hunk 1 above)
Comment (Mikearaya) [reply spec/iterator_spec.rb]: **Noted thanks**

File: spec/iterator_spec.rb (hunk 2 above)
Comment (Mikearaya) [reply spec/iterator_spec.rb]: **Noted thanks**

File: spec/iterator_spec.rb (hunk 3 above)
Comment (Mikearaya) [reply spec/iterator_spec.rb]: **Noted thanks**

File: spec/iterator_spec.rb (hunk 5 above)
Comment (Mikearaya) [reply spec/iterator_spec.rb]: **Removed**

File: spec/iterator_spec.rb (hunk 4 above)
Comment (Mikearaya) [reply spec/iterator_spec.rb]: **Removed**

File: spec/iterator_spec.rb (hunk 6 above)
Comment (Mikearaya) [reply spec/iterator_spec.rb:94]: **removed**

**Comment by Mikearaya** [comment]
Hi, @mosesogwo Thank you for your thorough review and recommendations. 

I have added the missing examples for `#my_any?` and `#my_all?` methods. I have also removed duplicate tests made for alias methods (`#my_inject`, `#my_find_all` & `#my_select`).

### Now I'm kindly requesting another review so I can merge to develop id all checks out.

**Thank you** :bowing_man:

**Review by maelfosso — APPROVED** [review]
# Congratulations :clap: 

It's OK :smile: You can merge it now :+1: 
Good continuation :handshake: 

Mael FOSSO
